core/DataAccess/ArchiveTableDao.php

```diff
@@ -86,4 +88,55 @@ public function getArchiveTableAnalysis($tableDate)
 
         return $result;
     }
+
+    /**
+     * Return invalidation queue table data
+     *
+     * @param bool $prettyTime
+     *
+     * @return array
+     * @throws \Exception
+     */
+    public function getInvalidationQueueData(bool $prettyTime = false): array
+    {
+        $invalidationsTable = Common::prefixTable("archive_invalidations");
+        $sql = "
+            SELECT 
+                *                
+            FROM `$invalidationsTable` 
+            ORDER BY ts_invalidated, idinvalidation asc";
+        $invalidations = Db::fetchAll($sql);
+
+        $metricsFormatter = new Formatter();
+
+        $data = [];
+        foreach ($invalidations as $i) {
+
+            $waiting = Date::now()->getTimestampUTC() - Date::factory($i['ts_invalidated'])->getTimestampUTC();
+            $processing = $i['ts_started'] ? Date::now()->getTimestampUTC() - (int) $i['ts_started'] : '';
+
+            if ($prettyTime) {
+                $waiting = $metricsFormatter->getPrettyTimeFromSeconds($waiting, true);
+                if ($processing != '') {
+                    $processing = $metricsFormatter->getPrettyTimeFromSeconds($processing, true);
+                }
+            }
+
+            $d = [];
+            $d['Invalidation'] = $i['idinvalidation'];
+            $d['Site'] = $i['idsite'];
+            $d['Period'] = ($i['period'] == 1 ? 'Day' : ($i['period'] == 2 ? 'Week' : ($i['period'] == 3 ? 'Month' :
+                ($i['period'] == 4 ? 'Year' : 'Range'))));
```

Comment:
Do we have constants somewhere for the periods? Or something like `PeriodPretty` or `PeriodToText` sort of a method?

Reply:
I looked something like this but couldn't find anything usable, the period name is a separate constant on each period class and the various getPeriodLabel() functions are expecting a period object to have been instantiated from the request parameters which use a string constant not the numeric value here.